### PR TITLE
WIP: Support "causal" padding in tf.layers.convolutional.Conv1D

### LINKED ISSUE #14933: Support "causal" padding in tf.keras.layers.Conv1D by adding support to tf.layers.convolutional.Conv1D
TensorFlow version:1.4.0

In the API docs, tf.keras.layers.Conv1D supports "causal". However, if I use "causal" as the argument of 'padding', there will be an error like this: The 'padding' argument must be one of" valid "," same ". Received : causal.

I check the source code, the Conv1D uses the methods in tensorflow.python.layers.convolutional, but the default implementations of tensorflow do not support  "causal" as the padding.

(p.s., fchollet/keras uses the conv1d method in tensorflow_backend.py)

### PR DESCRIPTION
**Pull request by facaiy**
Fix #14933.

Because we don't see causal padding in other use cases expect of NTC, we choose to modify code at Conv1D, instead of `tf.nn.convolution`.
For simplicity, we hack the Conv1D's `__call__` method, instead of `build`, `call` and `_compute_output_shape`.

### How to test

+ [x] add test case.
+ [ ]  pass all tests.

### CODE REVIEW

**Comment by martinwicke** [comment]
Lukasz, Francois, I don't know enough about needs in 1D convolutions to know if this is useful and correct. Could you take a look?

**Comment by facaiy** [comment]
Thanks, @martinwicke . As @szpssky mentioned, the PR cannot work for keras, so I mark the PR as WIP.

**Comment by martinwicke** [comment]
Ok, thanks!

**Comment by lukaszkaiser** [comment]
In general, it'd be very very good to have causal="LEFT" padding in TF. This is the one reason why both T2T and Sonet re-implement conv layers...
